### PR TITLE
WIP/CI: use windows-2022

### PR DESCRIPTION
**Pull request by pheiduck**
WIP PR to get the CI ready for Windows 2022

### CODE REVIEW

**Comment by pheiduck** [comment]
Open Task:
- Get Python2 to install
- Get Build working again on mingw and msvc

Would be great if someone help out.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -422,7 +422,7 @@ jobs:
         shell: bash
         run: |
           git config --global core.autocrlf input
-          echo "VCVARSALL=$(vswhere -products \* -latest -property installationPath)\\VC\\Auxiliary\\Build\\vcvarsall.bat" >> $GITHUB_ENV
+          echo "VCVARSALL=$(vswhere -products \* -2019 -property installationPath)\\VC\\Auxiliary\\Build\\vcvarsall.bat" >> $GITHUB_ENV
           python_dir=$(cat "/proc/${{ matrix.cygreg }}/HKEY_LOCAL_MACHINE/SOFTWARE/Python/PythonCore/${PYTHON_VER_DOT}/InstallPath/@")
           python3_dir=$(cat "/proc/${{ matrix.cygreg }}/HKEY_LOCAL_MACHINE/SOFTWARE/Python/PythonCore/${PYTHON3_VER_DOT}${{ matrix.pyreg }}/InstallPath/@")
```
Comment (k-takata) [inline .github/workflows/ci.yml]: These two lines are used to read the python's install path from the registry.
If you install python to a directory you defined, you can directly set the environment variable `PYTHON(3)_DIR` by yourself.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -422,8 +422,9 @@ jobs:
         shell: bash
         run: |
           git config --global core.autocrlf input
-          echo "VCVARSALL=$(vswhere -products \* -2019 -property installationPath)\\VC\\Auxiliary\\Build\\vcvarsall.bat" >> $GITHUB_ENV
-          python_dir=$(cat "/proc/${{ matrix.cygreg }}/HKEY_LOCAL_MACHINE/SOFTWARE/Python/PythonCore/${PYTHON_VER_DOT}/InstallPath/@")
+          echo "VCVARSALL=$(vswhere -products \* -latest -property installationPath)\\VC\\Auxiliary\\Build\\vcvarsall.bat" >> $GITHUB_ENV
+          choco install python2 --params "/InstallDir:C:\"
+          python_dir=$(cat "C:\")
           python3_dir=$(cat "/proc/${{ matrix.cygreg }}/HKEY_LOCAL_MACHINE/SOFTWARE/Python/PythonCore/${PYTHON3_VER_DOT}${{ matrix.pyreg }}/InstallPath/@")
           echo "PYTHON_DIR=$python_dir" >> $GITHUB_ENV
```
Comment (k-takata) [inline .github/workflows/ci.yml]: You can directly define an environment variable around line 373:
```yaml
      # Python 2
      PYTHON_VER: 27
      PYTHON_VER_DOT: '2.7'
      PYTHON_DIR: 'C:\Python27'
```

Then, you don't need to set the variable `python_dir` and you can remove the line `echo "PYTHON_DIR=$python_dir" >> $GITHUB_ENV`.

Or, the `choco install python2` command properly sets the registry, you should be able to leave the `python_dir=$(cat "/proc/ ... )` command as is.

**Comment by pheiduck** [comment]
ToDo's:
- rebase into 1 commit
- fix Builds with mingw

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -424,9 +425,8 @@ jobs:
           git config --global core.autocrlf input
           echo "VCVARSALL=$(vswhere -products \* -latest -property installationPath)\\VC\\Auxiliary\\Build\\vcvarsall.bat" >> $GITHUB_ENV
           choco install python2
-          python_dir="C:\Python27"
           python3_dir=$(cat "/proc/${{ matrix.cygreg }}/HKEY_LOCAL_MACHINE/SOFTWARE/Python/PythonCore/${PYTHON3_VER_DOT}${{ matrix.pyreg }}/InstallPath/@")
-          echo "PYTHON_DIR=$python_dir" >> $GITHUB_ENV
+          echo "PYTHON_DIR" >> $GITHUB_ENV
```
Comment (k-takata) [inline .github/workflows/ci.yml]: This line is not needed.

**Comment by k-takata** [comment]
Another thing you need to care about is that you should install the 32-bit version of Python 2 when you test 32-bit version of Vim.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -423,14 +424,16 @@ jobs:
...
           python3_dir=$(cat "/proc/${{ matrix.cygreg }}/HKEY_LOCAL_MACHINE/SOFTWARE/Python/PythonCore/${PYTHON3_VER_DOT}${{ matrix.pyreg }}/InstallPath/@")
-          echo "PYTHON_DIR=$python_dir" >> $GITHUB_ENV
           echo "PYTHON3_DIR=$python3_dir" >> $GITHUB_ENV
 
       - uses: msys2/setup-msys2@v2
         if: matrix.toolchain == 'mingw'
         with:
+          update: true
+          install: >-
+            mingw-w64-x86_64-make gcc
```
Comment (k-takata) [inline .github/workflows/ci.yml]: You are installing the wrong type of gcc.
```suggestion
          pacboy: >-
            make:p gcc:p
```

**Pull request closed without merging**